mkosi

```diff
@@ -3498,6 +3512,34 @@ def run_qemu(args):
 
     os.execvp(cmdline[0], cmdline)
 
+def expand_paths(input_paths):
+    if not input_paths:
+        return []
+
+    environ = os.environ.copy()
+    # Add a fake SUDO_HOME variable to allow non-root users specify
+    # paths in their home when using mkosi via sudo.
+    sudo_user = os.getenv("SUDO_USER")
+    if sudo_user:
+        environ["SUDO_HOME"] = os.path.expanduser("~{}".format(sudo_user))
+
+    # No os.path.expandvars because it treats unset variables as empty.
+    paths = []
+    for path in input_paths.split(":"):
+        try:
+            path = string.Template(path).substitute(environ)
+            paths.append(path)
+        except KeyError:
+            # Skip path if it uses a variable not defined.
+            pass
+    return paths
+
+def prepend_environ_path(paths):
+    real_path = os.getenv("PATH", None)
+    if real_path is not None:
+        paths.append(real_path)
```

Comment:
i don't like this tbh... append to this list a string that contains multiple colon-separate elements is just ugly...

I'd much prefer if we'd append the original PATH env var to the end of the new env var as a string, and never add it to the list if you follow what I mean... maybe something like this:

```python
if len(paths) == 0:
    return

original_path = os.getenv("PATH", None)
new_path = ":".join(paths)

if original_path is not None:
    environ["PATH"] = original_path + ":" + new_path
else
    environ["PATH"] = new_path
```

But maybe there's a more pythonic way to do this? @keszybz might be able to help?

Reply:
Done like your suggestion.